src/worker.cpp

```diff
@@ -1393,4 +1393,99 @@ void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
         */
     }
 }
+
+void Worker::deleteFruVpd(const std::string& i_dbusObjPath)
+{
+    try
+    {
+        if (m_parsedJson.empty())
+        {
+            throw std::runtime_error(
+                "System config JSON is empty. Cannot perform deletion for: " +
+                i_dbusObjPath);
+        }
+
+        const std::string& l_fruPath =
+            jsonUtility::getFruPathFromJson(m_parsedJson, i_dbusObjPath);
+
+        if (l_fruPath.empty())
+        {
+            throw std::runtime_error(
+                "DBus object path not present in system config JSON. Cannot perform deletion for: " +
+                i_dbusObjPath);
+        }
+
+        const bool l_presentPropValue =
+            std::get<bool>(dbusUtility::readDbusProperty(
+                constants::pimServiceName, i_dbusObjPath,
+                constants::inventoryItemInf, "Present"));
+
+        if (!l_presentPropValue)
+        {
+            logging::logMessage(
+                "Given FRU is not present. Could not perform deletion for " +
+                i_dbusObjPath);
+            return;
+        }
+        else
+        {
+            std::vector<std::string> l_interfaceList{
+                constants::operationalStatusInf};
+
+            types::MapperGetSubTree l_subTreeMap =
+                dbusUtility::getObjectSubTree(i_dbusObjPath, 0,
+                                              l_interfaceList);
+
+            types::ObjectMap l_objectMap;
+
+            // Updates the present and functional property for the sub FRU's.
+            for (const auto& [l_objectPath, l_serviceInterfaceMap] :
+                 l_subTreeMap)
+            {
+                // ToDo -- Update functional property to true.
+                types::InterfaceMap l_interfaceMap{
+                    {constants::inventoryItemInf, {{"Present", false}}}};
+                l_objectMap.emplace(l_objectPath, std::move(l_interfaceMap));
+            }
+
+            if (jsonUtility::isActionRequired(m_parsedJson,
```

Comment:
This should only be applicable if json is there else skip this part.
Basic Idea is can we design this API such that it can work with or without JSON?
I think it can be done and used in system without JSON just to clear data of a Dbus object.

Reply:
Updated to perform deletion with or without JSON